### PR TITLE
Experimental filters: fix providers issues

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
This PR fixes issues introduced in #3899. Mainly errors when fetching the initial data (hence extracting hook from API provider) and with persisting state of filter popover.

ref: https://github.com/saleor/saleor-dashboard/issues/3272
 
### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

**Review by andrzejewsky — APPROVED** [review]

File: src/components/ConditionalFilter/API/InitialStateResponse.ts [hunk 1]
```diff
@@ -31,6 +31,10 @@ export class InitialStateResponse implements InitialState {
     return this.attribute[name];
   }
 
+  public static empty() {
+    return new InitialStateResponse([], {}, [], [], []);
+  }
+
```
Comment (Droniu) [inline src/components/ConditionalFilter/API/InitialStateResponse.ts:37]: **suggestion**: InitialStateResponse has 5 parameters - this is unreadable, please use object parameter.

File: src/components/ConditionalFilter/API/initialState/useInitialAPIState.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import { createInitialStateFromData } from "./helpers";
+import { InitialAPIResponse } from "./types";
+
+export interface InitialAPIState {
+  data: InitialStateResponse;
+  loading: boolean;
+  fetchQueries: (params: FetchingParams) => void;
+}
+
+export const useProductInitialAPIState = (): InitialAPIState => {
```
Comment (poulch) [inline src/components/ConditionalFilter/API/initialState/useInitialAPIState.tsx:32]: question: Shouldn't be `useInitialAPIState`?

**Review by Droniu — APPROVED** [review]

**Review by poulch — APPROVED** [review]

File: src/components/ConditionalFilter/API/initialState/useInitialAPIState.tsx (hunk 2 above)
Comment (andrzejewsky) [reply src/components/ConditionalFilter/API/initialState/useInitialAPIState.tsx:32]: hmm don't think so, useInitialAPIState suggest type for entire hook, while here he types just the returned value

Comment (andrzejewsky) [reply src/components/ConditionalFilter/API/initialState/useInitialAPIState.tsx:32]: he wants to have initial state for products, perhaps file can be renamed/splited

**Pull request merged**